src/components/Modal/Modal.js

```diff
@@ -0,0 +1,126 @@
+import React, { useRef } from 'react'
+import PropTypes from 'prop-types'
+import { Portal } from '../Portal'
+import styles from './Modal.module.scss'
+import useOutsideClick from '../../hooks/a11y/useOutsideClick'
+import useBodyScrollLock from '../../hooks/a11y/useBodyScrollLock'
+import useTrapFocus from '../../hooks/a11y/useTrapFocus'
+import useHideAriaSiblings from '../../hooks/a11y/useHideAriaSiblings'
+
+/**
+ * Component renders the internal content of a modal
+ *
+ * @param {object} props - the component's props
+ * @param {React.ReactChildren} props.children - the component's children
+ * @param {(val: boolean) => any} props.toggle - callback function that toggles the
+ * show modal state
+ *
+ * @return {JSX.Element}
+ */
+const ModalContent = ({
+  ariaDescribedBy,
+  ariaLabelledBy,
+  children,
+  toggle,
+  showModal,
+}) => {
+  const modalRef = useRef(null)
+
+  // Hooks! All the thingggs!!!!!
+  useOutsideClick(modalRef, () => toggle(false))
+  useTrapFocus(modalRef, showModal)
+  useBodyScrollLock(showModal)
+  useHideAriaSiblings(modalRef, showModal)
+
+  return (
+    <div
+      role="dialog"
+      aria-modal="true"
+      aria-describedby={ariaDescribedBy && ariaDescribedBy}
+      aria-labelledby={ariaLabelledBy && ariaLabelledBy}
+      className={styles.ModalContent}
+      ref={modalRef}
+    >
+      {children}
+    </div>
+  )
+}
+
+ModalContent.propTypes = {
+  ariaDescribedBy: PropTypes.string,
```

Comment:
Which of these are required vs. optional? I think `isRequired` my be needed for a couple of these

Reply:
`children`, `toggle`, and `showModal` are the only required props for `ModalContent`. I'm not sure how we want to handle these cases as `ModalContent` is an internal implementation of the `Modal` and not exposed as part of the public API

Reply:
Ah gotcha, then maybe just remove the PropTypes for the internal ModalContent

Reply:
I think it's fine to leave them for now, I'd rather not disable the eslint rule for that block. This should be a mute point when we move completely to typescript 🙃